provider/examples/resources/asa/asav/variables.tf

```diff
@@ -0,0 +1,86 @@
+variable "vpc_id" {
+  description = "The AWS VPC to deploy the ASAv in."
+  type        = string
+}
+
+variable "base_name" {
+  description = "The base name is a prefix assigned to all of the resources created for the ASAv. It typically corresponds to the name of the CDO environment that the ASAv will be onboarded to."
+  type        = string
+}
+
+variable "public_subnets" {
+  description = "The ASAv is deployed with three interfaces (management, outside, and inside) to a AWS VPC with public subnets accessible from the internet, and private subnets inaccessible from the internet. The management and outside interfaces of the ASAv are deployed in the public subnet. Specify the public subnets in the VPC here."
+  type        = list(any)
+  validation {
+    condition     = length(var.public_subnets) >= 1
+    error_message = "The ASAv requires at least one subnet that the management and outside interfaces can be deployed to."
+  }
+}
+
+variable "private_subnets" {
+  description = "The ASAv is deployed with three interfaces (management, outside, and inside) to a AWS VPC with public subnets accessible from the internet, and private subnets inaccessible from the internet. The management and outside interfaces of the ASAv are deployed in the public subnet. Specify the private subnets in the VPC here."
+  type        = list(any)
```

Comment:
Should `public_subnets` and `private_subnets` be lists? I think we only ever use `[0]`